### PR TITLE
Adding the Solution of a problem 523

### PR DESCRIPTION
**Pull request by parikhitritgithub**
## Related Issue

..

## Description

Adding the Solution of a problem 523
issue :  #890

## Type of PR

- [ ] Bug fix
- [ ] Feature enhancement
- [ ] Documentation update
- [ ] Other (specify): _______________

## Checklist
- [ ] I have performed a self-review of my code.
- [ ] I have read and followed the Contribution Guidelines.
- [ ] I have tested the changes thoroughly before submitting this pull request.
- [ ] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [ ] I have commented my code, particularly in hard-to-understand areas.

## Resources for Guidance

Here are some resources that may be helpful as you contribute to CodeHarborHub:
- [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [Markdown Guide](https://www.markdownguide.org/)
- [MDX Documentation](https://mdxjs.com/docs/)
- [Mermaid Documentation](https://mermaid.js.org/)

**Resources for DSA Features on Our Site:**

- Tabs: [Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex)

### CODE REVIEW

File: dsa-solutions/lc-solutions/0500-0599/523. Continuous Subarray Sum.md [hunk 1]
```diff
@@ -0,0 +1,211 @@
+---
+id: Continuous Subarray Sum
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0500-0599/523. Continuous Subarray Sum.md:2]: replace `id: Continuous Subarray Sum` to `id: continuous-subarray-sum`

File: dsa-solutions/lc-solutions/0500-0599/523. Continuous Subarray Sum.md [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+```
+Input: nums : [23,2,6,4,7], k : 13
+Output: false
+```
+
+
+### Constraints
+
+-  $1 <= nums.length <= 105$
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0500-0599/523. Continuous Subarray Sum.md:75]: replace `$1 <= nums.length <= 105$` to `$1 \leq \text{nums.length} \leq  105$`

File: dsa-solutions/lc-solutions/0500-0599/523. Continuous Subarray Sum.md [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+### Intuition
+This is a prefix sum's problem. Due to LC
+
+    A good subarray is a subarray where:
+    its length is at least two, and
+    the sum of the elements of the subarray is a multiple of k.
+
+modulo k there are 0,1,...k-1 totally k possible for prefix sum (mod k).
+For this constraint $1 <= nums.length <= 10^5$ an O(n^2) solution may lead to TLE.
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0500-0599/523. Continuous Subarray Sum.md:90]: `O(n^2)` to `$O(n^2)$`

File: dsa-solutions/lc-solutions/0500-0599/523. Continuous Subarray Sum.md [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+A hash map with care on prefix sum mod k to use is however a tip. The array version is used when k<n combining the hash map version that will be the faster than other solutions.
+
+Thanks to the comments of @Sergei proposing to use unordered_set, a combination of bitset with unordered_set is implemented which outperforms all other solutions with the elapsed time 86ms & beats 99.75%
+
+
+### Approach
+
+
+   - First try uses array version for mod_k(k), but LC's constraints: $1 <= k <= 2^31 - 1$ which leads to MLE.
+   - 2nd approach uses unordered_map<int, vector<int>> instead of an array which is accepted by LC.
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0500-0599/523. Continuous Subarray Sum.md:102]: `unordered_map<int, vector<int>>` to 

```md
`unordered_map<int, vector<int>>`
```

File: dsa-solutions/lc-solutions/0500-0599/523. Continuous Subarray Sum.md [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+   - Since the computation uses mod_k[prefix].front(), a simple hash table unordered_map<int, int> mod_k is sufficient for this need.
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0500-0599/523. Continuous Subarray Sum.md:103]: `unordered_map<int, int> mod_k` 

```md
`unordered_map<int, int> mod_k`
```

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
update code based on suggestions

File: dsa-solutions/lc-solutions/0500-0599/523. Continuous Subarray Sum.md [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+
+    A good subarray is a subarray where:
+    its length is at least two and
+    the sum of the elements of the subarray is a multiple of k
+
+modulo k there are 0,1,...k-1 totally k possible for prefix sum (mod k)
+For this constraint $1 \leq \text{nums.length} \leq  105$ an $O(n^2)$ solution may lead to TLE
+
+
+A hash map with care on prefix sum mod k to use is however a tip  The array version is used when k<n combining the hash map version that will be the faster than other solutions
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0500-0599/523. Continuous Subarray Sum.md:93]: replace `k<n` to

```md
`k<n`
```

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
now solve your last error and complete task

File: dsa-solutions/lc-solutions/0500-0599/523. Continuous Subarray Sum.md (hunk 6 above)
Comment (parikhitritgithub) [reply dsa-solutions/lc-solutions/0500-0599/523. Continuous Subarray Sum.md]: finally done .

**Pull request closed without merging**